lib/segment/benches/dynamic_mmap_flags.rs

```diff
@@ -0,0 +1,61 @@
+use std::iter;
+use std::sync::atomic::AtomicBool;
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use rand::rngs::StdRng;
+use rand::{Rng, SeedableRng};
+use segment::common::operation_error::check_process_stopped;
+use segment::vector_storage::dense::dynamic_mmap_flags::DynamicMmapFlags;
+use tempfile::tempdir;
+
+const FLAG_COUNT: usize = 50_000_000;
+
+fn dynamic_mmap_flag_count(c: &mut Criterion) {
+    let mut rng = StdRng::seed_from_u64(42);
+    let dir = tempdir().unwrap();
+    let random_flags: Vec<bool> = iter::repeat_with(|| rng.gen()).take(FLAG_COUNT).collect();
+
+    // Build dynamic mmap flags with random deletions
+    let mut dynamic_flags = DynamicMmapFlags::open(dir.path()).unwrap();
+    dynamic_flags.set_len(FLAG_COUNT).unwrap();
+    random_flags
+        .iter()
+        .enumerate()
+        .filter(|(_, flag)| **flag)
+        .for_each(|(i, _)| assert!(!dynamic_flags.set(i, true)));
+    dynamic_flags.flusher()().unwrap();
+    let real_count = random_flags.iter().filter(|&&flag| flag).count();
+
+    let mut group = c.benchmark_group("dynamic-mmap-flag-count");
+
+    group.bench_function("manual-count-loop", |b| {
+        b.iter(|| {
+            let mut count = 0;
+            let stopped = AtomicBool::new(false);
```

Comment:
maybe you can remove the `AtomicBool` creation from the measurement to remove some noise

Reply:
Will do.

It is kind of useless in this test anyway; I only kept it there because it's also in our original counting logic.